### PR TITLE
OBPIH-6217 Setup user login

### PR DESCRIPTION
**Pull request by drodzewicz**
In this PR I have implemented all of the required logic for user login before executing further tests. 
Before all tests, a setup file is executed with authentication for provided users in the `.env` file.
I have also implemented the logic that allows us to test users with different roles like the example I have prepared, where one test is executed for our main test user and the other for the requestor (as a user with **requestor** role).

During planning me and Katarzyna came to the conclusion that it would be better to use multiple users for different scenarios instead of operating on a single user and changing it's roles before each test.

Additionally, in the setup file after login in we want to perform validation on the provided users to make sure that this user has all of the expected roles. This way we can eliminate many false positive tests.
A case might happen where we expect the Test User to have certain roles but we provided the testing environment a user without that specific role, so instead of debugging later and trying to find the reason why test are failing I added a validation to instantly point us in the right direction.

### CODE REVIEW

File: src/api/GenericService.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+class GenericService {
+  private request: APIRequestContext;
+
+  constructor(request: APIRequestContext) {
+    this.request = request;
+  }
+
+  async getAppContext() {
+    const apiRepsonse = await this.request.get('./api/getAppContext');
```
Comment (alannadolny) [inline src/api/GenericService.ts:11]:
```suggestion
    const apiResponse = await this.request.get('./api/getAppContext');
```

File: src/utils/index.ts
[the remark this answers is not in the document]
Comment (alannadolny) [reply src/utils/index.ts:1]: Lodash won't be helpful in that case?

File: src/utils/TestUser.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    this.storagePath = path.join(
+      AppConfig.AUTH_STORAGE_DIR_PATH,
+      storageFileName
+    );
+    this.requiredRoles = requiredRoles;
+  }
+
+  assertAllRequiredRoles(userRoles: Set<string>) {
+    const unexpectedRoles = _.difference([...userRoles], [...this.requiredRoles]);
+    const absentRequiredRoles = _.difference([...this.requiredRoles], [...userRoles]);
```
Comment (alannadolny) [inline src/utils/TestUser.ts:31]: Are those spread operators necessary?

Comment (drodzewicz) [reply src/utils/TestUser.ts:31]: `userRole` and `requiredRoles` are of type `Set` and `_.difference()` takes in arrays

**Pull request merged**